tests/223_require_execute_match_empty.test

```diff
@@ -0,0 +1,37 @@
+#!/bin/sh
+
+
+# Tests that the "require-execute-match" feature works. This
+# allows using the execute escape hatch for require-conditions
+
+. "$(cd "$(dirname "$0")" && pwd)/common.sh"
+
+cat >$CONFIG <<EOF
+task upgrade.fail_empty_1 {
+    require-execute-match("echo \"\"", "foo")
+    on-init { info("incorrect") }
+}
+task upgrade.fail_empty_2 {
+    require-execute-match("echo foo", "")
+    on-init { info("incorrect") }
+}
+task upgrade.match_empty {
+    require-execute-match("echo \"\"", "")
```

Comment:
Instead of `echo` for this test, I think that `true` and `false` would be good to try:

```suggestion
    # Probably count as success even though program returns an error
    require-execute-match("false", "")
```

```suggestion
    # Shouldn't work. Program not existing should never work
    require-execute-match("does_not_exist", "")
```

```suggestion
    # Should work
    require-execute-match("true", "")
```

Reply:
Ah, good ones :)

This makes me think it would make sense to have `require-execute-status` or something as well but I think we can hold off on that.